src/core.c/Range.rakumod

```diff
@@ -57,10 +57,12 @@ my class Range is Cool does Iterable does Positional {
         nqp::create(self)!SET-SELF(-Inf,Inf,$excludes-min,$excludes-max,1);
     }
     multi method new(Whatever \min, \max, :$excludes-min, :$excludes-max) {
-        nqp::create(self)!SET-SELF(-Inf,max,$excludes-min,$excludes-max,1);
+        use fatal;
```

Comment:
Hmmm.. but then why the "use fatal".  I seem to be missing something?

Reply:
Because otherwise max.Real as a Failure will simply be passed on, and a monstrous Range (that starts/ends with a Failure) will be created.

Reply:
Aha, ok, then maybe SET-SELF would need an `$max.throw if nqp::istype($max,Failure)` ?

Reply:
Probably it should check for both $min and $max; I'm not sure of the performance implications. Right now, the other cases only throw because they tried to retrieve the failed value before doing the function call.